yaaredis/pool.py

```diff
@@ -647,3 +648,235 @@ def get_node_by_slot(self, slot):
+                 skip_full_coverage_check=False, nodemanager_follow_cluster=False, readonly=False,
+                 max_idle_time=0, idle_check_interval=1,
+                 **connection_kwargs):
+        """
+        :skip_full_coverage_check:
+            Skips the check of cluster-require-full-coverage config, useful for clusters
+            without the CONFIG command (like aws)
+        :nodemanager_follow_cluster:
+            The node manager will during initialization try the last set of nodes that
+            it was operating on. This will allow the client to drift along side the cluster
+            if the cluster nodes move around alot.
+        """
+        super().__init__(
+            connection_class=connection_class, max_connections=max_connections)
+
+        assert max_idle_time == 0, 'Max idle time not support for blocking version'
+        assert max_connections_per_node, 'Only per node connection limit is supported'
+
+        # Special case to make from_url method compliant with cluster setting.
+        # from_url method will send in the ip and port through a different variable then the
+        # regular startup_nodes variable.
+        if startup_nodes is None:
+            if 'port' in connection_kwargs and 'host' in connection_kwargs:
+                startup_nodes = [{
+                    'host': connection_kwargs.pop('host'),
+                    'port': str(connection_kwargs.pop('port')),
+                }]
+
+        self.max_connections = max_connections or 2 ** 31
+        self.nodes = NodeManager(
+            startup_nodes,
+            reinitialize_steps=reinitialize_steps,
+            skip_full_coverage_check=skip_full_coverage_check,
+            max_connections=self.max_connections,
+            nodemanager_follow_cluster=nodemanager_follow_cluster,
+            **connection_kwargs,
+        )
+        self.initialized = False
+
+        self.connections = {}
+        self.connection_kwargs = connection_kwargs
+        self.connection_kwargs['readonly'] = readonly
+        self.readonly = readonly
+        self.max_idle_time = max_idle_time
+        self.idle_check_interval = idle_check_interval
+        self.reset()
+
+        if 'stream_timeout' not in self.connection_kwargs:
+            self.connection_kwargs['stream_timeout'] = ClusterConnectionPool.RedisClusterDefaultTimeout
+
+    def __repr__(self):
+        """
+        Returns a string with all unique ip:port combinations that this pool
+        is connected to
+        """
+        return '{}<{}>'.format(
+            type(self).__name__,
+            ', '.join([self.connection_class.description.format(**node)
+                       for node in self.nodes.startup_nodes]),
+        )
+
+    async def initialize(self):
+        if not self.initialized:
+            await self.nodes.initialize()
+            self.initialized = True
+
+    def _make_connection_queue(self):
+        res = asyncio.LifoQueue(maxsize=self.max_connections)
```

Comment:
The non-cluster implementation of a blocking client lets users pass in the queue implementation (see the `queue_class` handling) -- it'd be great to get the same behaviour here!